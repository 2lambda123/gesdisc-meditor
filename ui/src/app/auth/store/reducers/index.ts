import { createSelector, createFeatureSelector } from '@ngrx/store';
import * as fromAuth from './auth';

export interface AuthState {
  status: fromAuth.State;
}


export const reducers = {
  status: fromAuth.reducer
};

export const selectAuthState = createFeatureSelector<AuthState>('auth');

export const selectAuthStatusState = createSelector(
  selectAuthState,
  (state: AuthState) => state.status
);

export const getLoggedIn = createSelector(
  selectAuthStatusState,
  fromAuth.getLoggedIn
);

export const getUser = createSelector(
  selectAuthStatusState, 
  fromAuth.getUser
);