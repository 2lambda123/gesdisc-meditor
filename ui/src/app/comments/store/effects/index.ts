import { CommentsEffects } from './comments.effects';

export const effects: any[] = [CommentsEffects];

export * from './comments.effects';