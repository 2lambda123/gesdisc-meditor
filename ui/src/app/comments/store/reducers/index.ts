import {
	createFeatureSelector,
	ActionReducerMap,
} from '@ngrx/store';
import * as fromComments from './comments.reducer';

export interface CommentsState {
	comments: fromComments.State;
}

export const reducers: ActionReducerMap<CommentsState> = {
	comments: fromComments.reducer
};

export const getCommentsState = createFeatureSelector<CommentsState>('comments');



