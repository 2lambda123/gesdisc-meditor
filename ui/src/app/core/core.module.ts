import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
