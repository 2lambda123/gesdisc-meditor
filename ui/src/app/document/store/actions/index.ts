export * from './history.actions';
export * from './document.actions';