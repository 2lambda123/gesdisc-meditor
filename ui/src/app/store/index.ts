export { DocumentState } from './document/document.state';
export { ModelState } from './model/model.state';
export { NotificationState } from './notification/notification.state';
export { AuthState } from './auth/auth.state';
export { RouterState } from './router/router.state';
