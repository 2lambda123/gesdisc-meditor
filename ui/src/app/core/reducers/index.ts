import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
