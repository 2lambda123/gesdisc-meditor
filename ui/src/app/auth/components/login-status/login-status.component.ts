import { Component } from '@angular/core'
import { UserStore } from '../../../store'

@Component({
    selector: 'med-login-status',
    template: `
        <button
            mat-button
            (click)="logout()"
            color="accent"
            *ngIf="userStore.user$ | async as user"
            (mouseenter)="toggleButton()"
            (mouseleave)="toggleButton()"
        >
            <mat-icon>{{ userBtn ? 'person' : 'exit_to_app' }}</mat-icon>
            {{ userBtn ? 'Hi, ' + user.firstName : 'Logout' }}
        </button>
    `,
})
export class LoginStatusComponent {
    userBtn: boolean = true

    constructor(public userStore: UserStore) {}

    logout() {
        this.userStore.logout()
    }

    toggleButton() {
        this.userBtn = !this.userBtn
    }
}
