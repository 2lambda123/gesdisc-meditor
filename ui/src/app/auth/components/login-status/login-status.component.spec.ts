import { TestBed, ComponentFixture } from '@angular/core/testing';
import { Store, NgxsModule} from '@ngxs/store';
import { MatIconModule } from '@angular/material';
import { LoginStatusComponent } from './login-status.component';
import { of } from 'rxjs';

describe('Login Component', () => {
	let fixture: ComponentFixture<LoginStatusComponent>;
	let store: Store;
	let component: LoginStatusComponent;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [
				NgxsModule.forRoot(),
				MatIconModule
			],
			declarations: [LoginStatusComponent]
		});

		fixture = TestBed.createComponent(LoginStatusComponent);
		component = fixture.componentInstance;
		store = TestBed.get(Store);

		Object.defineProperty(component, 'user$', { writable: true });
		component.user$ = of({firstName: 'testName'});
		fixture.detectChanges();
	});

	it('should compile login status component', () => {
		expect(fixture).toMatchSnapshot();
	});

	it('userBtn should be true on init and should toggle userBtn status on toggleButton()', () => {
		expect(component.userBtn).toBeTruthy();
		component.toggleButton();
		expect(component.userBtn).toBeFalsy();
	});
});
