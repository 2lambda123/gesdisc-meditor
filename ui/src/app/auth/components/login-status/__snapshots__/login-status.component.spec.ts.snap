// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should compile login status component 1`] = `
<med-login-status
  store={[Function Store]}
  userBtn={[Function Boolean]}
>
  <button
    color="accent"
    mat-button=""
  >
    <mat-icon
      aria-hidden="true"
      class="mat-icon material-icons"
      role="img"
    >
      person
    </mat-icon>
     Hi, testName 
  </button>
</med-login-status>
`;
